share/man/man7/hier.7

```diff
@@ -25,432 +28,747 @@
+User utilities fundamental to both single-user and multi-user environments.
 .It Pa /boot/
-programs and configuration files used during bootstrap of the operating system
+Programs and configurations used during operating system bootstrap:
 .Pp
-.Bl -tag -width "nvmecontrol/" -compact
+.Bl -tag -width 14n -compact
 .It Pa defaults/
-default bootstrap configuration files; see
-.Xr loader.conf 5
+Default bootstrapping configuration files; see
+.Xr loader.conf 5 .
 .It Pa dtb/
-compiled flattened device tree (FDT) files; see
-.Xr fdt 4
+Compiled flattened device tree (FDT) files; see
+.Xr fdt 4 .
 and
-.Xr dtc 1
-.Pp
-.Bl -tag -width "overlays/" -compact
-.It Pa overlays/
-compiled flattened device tree (FDT) overlays; these can be applied using the
-.Va fdt_overlays
-setting in
-.Xr loader.conf 5
-.El
-.Pp
+.Xr dtc 1 .
 .It Pa efi/
-mount point for the EFI System Partition (ESP) on UEFI systems
+Mount point for EFI System Partition (ESP) on UEFI systems.
 .It Pa firmware/
-loadable kernel modules containing binary firmware, for hardware to which
-firmware must be downloaded
+Loadable kernel modules containing binary firmware for hardware that needs
+firmware downloaded to it to function.
+.It Pa fonts/
+Binary bitmap console fonts; see
+.Xr loader.conf 5
+and
+.Xr vtfontcvt 8 .
+.It Pa images/
+Beastie boot menu images; see
+.Xr loader_lua 8 .
 .It Pa kernel/
-pure kernel executable (the operating system loaded into memory at boot time)
-and kernel modules
+Pure kernel executable (the operating system loaded into memory
+at boot time) and kernel modules; see
+.Xr kldstat 8 .
+.It Pa kernel.old/
+Alternative kernel and modules.
+.It Pa loader.conf.d/
+.Xr loader.conf 5
+Configuration files.
+.It Pa lua/
+Scripts for kernel bootstrapping final stage; see
+.Xr loader_lua 8 .
 .It Pa modules/
-third-party loadable kernel modules, such as those associated with
-.Xr ports 7
+Third-party loadable kernel modules, such as the ones installed with
+.Xr pkg 8
+or from
+.Xr ports 7 .
 .It Pa zfs/
-.Xr zfs 8
-pool cache files
+ZFS
+.Xr zpool 8
+cache files.
 .El
 .It Pa /compat/
-files supporting binary compatibility with other operating systems
+Files needed to support binary compatibility with other operating systems:
 .Pp
-.Bl -tag -width "nvmecontrol/" -compact
+.Bl -tag -width 12n -compact
 .It Pa linux/
-default location for
+Default location for
 .Xr linux 4
-compatibility run-time
+compatibility run-time.
 .El
 .Pp
 .It Pa /dev/
-the normal mount point for
-.Xr devfs 5
+Device special files managed by
+.Xr devfs 5 :
 .Pp
-.Bl -tag -width "nvmecontrol/" -compact
+.Bl -tag -width 14n -compact
+.It Pa dri/
+GPU character device node; see
+.Xr drm 7 .
+.It Pa drm/
+No one knows exactly, but see above.
 .It Pa fd/
-file descriptor files; see
-.Xr fd 4
+File descriptor files;
+see
+.Xr \&fd 4 .
+.It Pa pts/
+Pseduo-terminals.
+.It Pa usb/
+USB busses.
 .El
+.It Pa /entropy
+Stored entropy file; see _____.
```

Comment:
The entropy file is used to provide initial state to the random number generator.
It is created and used by /etc/rc.d/random.
Depending on settings in /etc/rc.conf it may also be moved into some other directory or can be provided in a special place for the first boot (e.g. to make sure individual SSH host keys are generated on a freshly installed system or VM).
See random_start() in /etc/rc.d/random for details ...

Reply:
Awesome, thank you so much. Is there a knob for that we can mention in rc.conf(5)?
I feel like it'd be consistent to put the entropy file in /var.

Reply:
> Is there a knob for that we can mention in rc.conf(5)? I feel like it'd be consistent to put the entropy file in /var.

Providing entropy to the random generator supports many scenarios and therefore offers quite a number of options that can be configured in rc.conf:
```
entropy_boot_file="/boot/entropy"	# Set to NO to disable very early
			# (used at early boot time) entropy caching through reboots.
entropy_file="/entropy"	# Set to NO to disable late (used when going multi-user)
			# entropy through reboots.
			# /var/db/entropy-file is preferred if / is not avail.
entropy_dir="/var/db/entropy" # Set to NO to disable caching entropy via cron.
```
The comments explain the reasons for 3 different locations:

- _entropy_boot_file_ may be on a separate boot file system (not necessarily on the same partition as the root file system), it could for example also be provided by a network boot server
- _entropy_file_ is on the root file system and thus available before any user processes that need random numbers have been started (e.g. the initial start of sshd will need entropy to create an unpredictable host key)
- _entropy_dir_ contains additional entropy files with random data saved a few times per hour while the system is up and running (but /var may be on a file system different from the root file system and thus possibly not available at the start of going multi-user)

The contents of the _entropy_file_ and _entropy_boot_file_ will be overwritten with random data when going multi-user. An additional entropy file named /var/db/entropy-file is hard-coded in the /etc/rc.d/random. It will be read in if present (in addition to _entropy_file_), but will not be overwritten with new random data.

If modified, the value assigned to _entropy_boot_file_ in rc.conf needs to be added to loader.conf, too, since that file is read in by the loader and then passed to the kernel (see the definitions and comments in /boot/defaults/loader.conf):
```
# See rc.conf(5). The entropy_boot_file config variable must agree with the
# settings below.
entropy_cache_load="YES"		# Set this to NO to disable loading
					# cached entropy at boot time
entropy_cache_name="/boot/entropy"	# Set this to the name of the file
```